### PR TITLE
chore(weave): Enable LLM Judge creation from UI

### PR DESCRIPTION
**Pull request by tssweeney**
Continued part of LLM Judge project: UI Creation.

Screenshot 2024-11-15 at 08 55 56

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx [hunk 1]
```diff
@@ -0,0 +1,887 @@
...
+                onChange={newValue => updateInternalPair(index, key, newValue)}
+                style={{
+                  marginBottom: '0px',
+                }}
+              />
+            )}
+          </Box>
+          <Button
+            size="small"
+            variant="quiet"
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx:605]: I think we like ghost better

**Review by gtarpenning — APPROVED** [review]
huge

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx [hunk 2]
```diff
@@ -0,0 +1,887 @@
...
+      value={{
+        value: selectedValue,
+        label: selectedValue,
+      }}
+      onChange={v => updateConfig(targetPath, v.value, config, setConfig)}
+      style={{
+        marginBottom: noMarginBottom ? '0px' : GAP_BETWEEN_ITEMS_PX + 'px',
+      }}
+    />
+    // <FormControl
```
[the remark this answers is not in the document]
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx:436]: ty

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/baseObjectClassQuery.ts [hunk 3]
```diff
@@ -100,7 +100,7 @@ export const useCreateBaseObjectInstance = <
     createBaseObjectInstance(client, baseObjectClassName, req);
 };
 
-const createBaseObjectInstance = async <
+export const createBaseObjectInstance = async <
```
[the remark this answers is not in the document]
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/baseObjectClassQuery.ts:103]: yup!

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx (hunk 1 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx:605]: ack

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx [hunk 4]
```diff
@@ -0,0 +1,887 @@
...
+    updateConfig(
+      targetPath,
+      currentArray.filter((_: any, i: number) => i !== index),
+      config,
+      setConfig
+    );
+  }
+};
+
+const NumberField: React.FC<{
```
[the remark this answers is not in the document]
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx:700]: what do you mean?

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx [hunk 5]
```diff
@@ -0,0 +1,887 @@
...
+          keyName={key}
+          fieldSchema={fieldSchema as z.ZodTypeAny}
+          config={config}
+          setConfig={setConfig}
+          path={path}
+        />
+      ));
+    } else {
+      console.error('Unsupported schema type', configSchema);
+      return <Typography color="error">Unsupported schema type</Typography>;
```
[the remark this answers is not in the document]
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx:882]: Basically this entire thing was written with MUI, then i converted just the parts i needed to our stuff. I'll try to remove more mui stuff.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (gtarpenning) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx:700]: Although not in the create view, an edit flow might want to disable certain fields, like the name for example.

[the remark this answers is not in the document]
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ScorersPage/ZodSchemaForm.tsx:678]: Ahhh, yeah, makes sense. I haven't considered using this for an edit experience yet. Disable will be cool to add across all fields

**Pull request merged**